bin/problem.py

```diff
@@ -120,7 +142,28 @@ def __init__(self, yamldata: dict[str, Any], legacy_time_limit: Optional[float]
         )
         if not self.is_legacy() and self.problem_format_version != "2023-07-draft":
             fatal(f"problem_format_version {self.problem_format_version} not supported")
-        # TODO: also support 'type'. For now, we only support legacy 'validation'.
+
+        if self.is_legacy():
+            mode = parse_legacy_validation(parse_setting(yamldata, "validation", "default"))
+        else:
+            if "validation" in yamldata:
+                warn(
+                    "problem.yaml: 'validation' is removed in 2023-07-draft, please use 'type' instead"
+                )
+            mode = set(parse_setting(yamldata, "type", "pass-fail").split(" "))
+        self.interactive: bool = "interactive" in mode
+        self.multi_pass: bool = "multi-pass" in mode
+        self.custom_output: bool = (
+            self.interactive
+            or self.multi_pass
+            or (
+                "custom" in mode
+                if self.is_legacy()
+                # TODO: output_validator should be singular, but DOMjudge does not support this yet, so this should be fixed during export.
+                else (problem.path / "output_validators").exists()
```

Comment:
I don't think we want to support both plural and singular (`output_validators/` and `output_validator/`) right now? DOMjudge definitely doesn't support the latter (https://github.com/DOMjudge/domjudge/blob/main/webapp/src/Service/ImportProblemService.php#L932), and I doubt that Kattis does (at least Problemtools doesn't yet: https://github.com/Kattis/problemtools/blob/develop/problemtools/verifyproblem.py#L1465-L1466) :stuck_out_tongue:

Reply:
Following up in #424 so we can merge this PR for now :slightly_smiling_face: